sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/AggregationIterator.scala

```diff
@@ -181,11 +182,7 @@ abstract class AggregationIterator(
         // Process all expression-based aggregate functions.
         updateProjection.target(currentBuffer)(joinedRow(currentBuffer, row))
         // Process all imperative aggregate functions.
-        var i = 0
-        while (i < updateFunctions.length) {
-          updateFunctions(i)(currentBuffer, row)
-          i += 1
-        }
+        updateFunctions.foreach(updateFunction => updateFunction(currentBuffer, row))
```

Comment:
Thanks for your input @HyukjinKwon! I avoided the zipWithIndex because of that reason and kept the while loops unchanged. Just cached the length due to specified reason.

However for the parts that we don't need the index, we can use foreach method. It's not same as what is posted in [2]. 
The problem in [2] is `for (i <- Range) ` will be converted to `Range.foreach`. However in this particular case, we don't create the Range object and I don't think there is any virtual function calls. Please correct me if I am wrong.